src/cache/cacher.rs

```diff
@@ -41,10 +43,9 @@ impl RedisCache {
     /// # Arguments
     ///
     /// * `url` - It takes an url as a string.
-    pub fn cached_results_json(self, url: String) -> Result<String, Box<dyn std::error::Error>> {
-        let hashed_url_string = self.clone().compute_url_hash(&url);
-        let mut redis_connection: Connection =
-            Client::open(self.redis_connection_url)?.get_connection()?;
+    pub fn cached_results_json(&self, url: &str) -> Result<String, Box<dyn std::error::Error>> {
+        let hashed_url_string = Self::compute_url_hash(url);
+        let mut redis_connection: Connection = self.client.get_connection()?;
```

Comment:
Maybe we don't need to create a connection for each function, but we can create a connection once when creating `RedisCache`, I'm not familiar with redis, what do you think? @neon-mmd

Reply:
Yes, I think this is a great change :+1: because it reduces the number of times the connection between the server and client is established and disconnected which causes unnecessary waste of time and this change can improve the speed and performance of the app.